vendor/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/server/hooks.go

```diff
@@ -90,6 +90,15 @@ func (s *GenericAPIServer) AddPostStartHook(name string, hook PostStartHookFunc)
 	return nil
 }
 
+// AddPostStartHookOrDie allows you to add a PostStartHook, but dies on failure
+func (s *GenericAPIServer) AddPostStartHookOrDie(name string, hook PostStartHookFunc) error {
+	err := s.AddPostStartHook(name, hook)
+	if err != nil {
```

Comment:
`if err := s.AddPostStartHook(name, hook); err != nil {`

It is nice to have variables scoped to an if statement.